### PR TITLE
Add Gan folder,VanillaGAN-torch

### PR DESCRIPTION
**Pull request by Pramodh-G**

### CODE REVIEW

File: GAN/VanillaGAN-Torch/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+!git clone link-to-repo
+%run main.py --k_steps 5 --epoch 50 #customize according to what you want!
+```
+- on  a local machine :
+```bash
+git clone link-to-repo
+```
+```python
+python3 main.py --k_steps 5 --epoch 50 # customize as you want!
+```
```
Comment (nirmal-suthar) [inline GAN/VanillaGAN-Torch/README.md]: format this part according to resnet README

File: GAN/VanillaGAN-Torch/dataloader.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import torchvision.transforms as transforms
+from torch.utils.data import DataLoader
+from torchvision import datasets
+
+def get_loader(batch_size,num_workers):
+
+    batch_size = 100
```
Comment (nirmal-suthar) [inline GAN/VanillaGAN-Torch/dataloader.py]: use batchsize argument instead of 100
```suggestion
```

File: GAN/VanillaGAN-Torch/dataloader.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    compose = transforms.Compose(
+            [transforms.ToTensor(),
+            transforms.Normalize((.5,), (.5,))
+            ])
+    data = datasets.MNIST(root='./Data', train=True, transform=compose, download=True)
+    data_loader = DataLoader(data, batch_size=batch_size, shuffle=True)
+    #num_batches = len(data_loader)
+
+    return data_loader
+
```
Comment (nirmal-suthar) [inline GAN/VanillaGAN-Torch/dataloader.py]: remove unnecessary comments and blank lines at the end

Comment (nirmal-suthar) [reply GAN/VanillaGAN-Torch/dataloader.py]: also use root as parser argument

File: GAN/VanillaGAN-Torch/main.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        img = generator(test_noise).reshape(-1,1,28,28).cpu().detach()
+        img_grid = torchvision.utils.make_grid(img)
+        images.append(img_grid)
+        #store loss
+        g_losses.append(g_error/i) #mean error 
+        d_losses.append(d_error/i) # mean error
+        #print metrics
+        print(f'Epoch {epoch}: g_loss: {(g_error/i):.8f} d_loss: {(d_error/i):.8f} time:{time.time()-start_time} seconds')      
+
+    return
```
Comment (nirmal-suthar) [inline GAN/VanillaGAN-Torch/main.py]:
```suggestion
 
```

Comment (nirmal-suthar) [reply GAN/VanillaGAN-Torch/main.py]: and remove stray lines

File: GAN/VanillaGAN-Torch/main.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+#loading/preprocessing
+data_loader = get_loader(optim_config['batch_size'],run_config['num_workers'])
+generator = Generator()
+generator = generator.to(device)
```
Comment (nirmal-suthar) [inline GAN/VanillaGAN-Torch/main.py]:
```suggestion
generator = Generator().to(device)
```

File: GAN/VanillaGAN-Torch/models.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+class Generator(nn.Module):
+
+  """
+  the generator in a GAN
+  """
+  def __init__(self):
+    super().__init__()
+
+    n_features = 100
```
Comment (nirmal-suthar) [inline GAN/VanillaGAN-Torch/models.py]: make latent space dimension as parser argument

**Comment by nirmal-suthar** [comment]
rename base to generative_model and VanillaGAN-torch to VanillaGAN_pytorch

File: GAN/VanillaGAN-Torch/models.py (hunk 6 above)
Comment (Pramodh-G) [reply GAN/VanillaGAN-Torch/models.py]: done

File: GAN/VanillaGAN-Torch/README.md (hunk 1 above)
Comment (Pramodh-G) [reply GAN/VanillaGAN-Torch/README.md]: what should i exactly do here?

File: GAN/VanillaGAN-Torch/dataloader.py (hunk 3 above)
Comment (Pramodh-G) [reply GAN/VanillaGAN-Torch/dataloader.py]: The root meaning the out directory?

File: generative_models/VanillaGAN_PyTorch/main.py [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    data = Variable(torch.zeros(size, 1))
+    return data.to(device)
+
+
+def noise(size):
+  n = Variable(torch.randn(size, model_config['latent_dim']))
+  return n.to(device)    
+
+
+
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_PyTorch/main.py]:
```suggestion
```

Comment (nirmal-suthar) [reply generative_models/VanillaGAN_PyTorch/main.py]: remove stray lines, everywhere

**Review by nirmal-suthar — APPROVED** [review]
can you compress gif file, and upload it again. Size of gif file is over 35 MB and try to keep supporting files below 10 MB.

File: GAN/VanillaGAN-Torch/README.md (hunk 1 above)
Comment (nirmal-suthar) [reply GAN/VanillaGAN-Torch/README.md]: https://github.com/pclubiitk/model-zoo/blob/master/image_classification/ResNet-PyTorch/README.md  similar to this

Comment (nirmal-suthar) [reply GAN/VanillaGAN-Torch/README.md]: https://github.com/pclubiitk/model-zoo/blob/master/image_classification/ResNet-PyTorch/README.md (similar to this)

File: generative_models/VanillaGAN_PyTorch/main.py [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+test_noise = noise(test_size)
+
+train(data_loader,discriminator,generator,epochs,criterion,d_optimizer,g_optimizer,test_noise)
+
+plot_loss(g_losses,d_losses,outdir)
+make_gif(generator,images,outdir)
+show_generator(generator,test_noise)
+
+
+
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_PyTorch/main.py]: stray lines

File: generative_models/VanillaGAN_PyTorch/eval.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import numpy as np
+import torchvision.transforms as transforms
+import torchvision
+import matplotlib.pyplot as plt
+import imageio
+import os
+
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_PyTorch/eval.py:7]: stray lines

**Review by nirmal-suthar — APPROVED** [review]

File: generative_models/VanillaGAN_PyTorch/eval.py (hunk 9 above)
Comment (Pramodh-G) [reply generative_models/VanillaGAN_PyTorch/eval.py:7]: no more than a maximum of one line has been allotted at meaningful places :)
